render/vnode.js

```diff
@@ -1,7 +1,7 @@
 "use strict"
 
 function Vnode(tag, key, attrs, children, text, dom) {
-	return {tag: tag, key: key, attrs: attrs, children: children, text: text, dom: dom, domSize: undefined, state: undefined, events: undefined, instance: undefined, skip: false}
+	return {tag: tag, key: key, attrs: attrs, children: children, text: text, dom: dom, domSize: undefined, state: undefined, _state: undefined, events: undefined, instance: undefined, skip: false}
```

Comment:
Leo benchmarked them, the cost was IIRC accetptable (more than having the hidden class change).

It may be nice to split this into two objects (public/private), as suggested by @isiahmeadows elsewhere, though most of these properties can be useful in user code as well... #1339 didn't have an extra property because of this...

`skip` and `events` are unambiguously private... So is `_state` (though you could implement weird state machines with it). I think @barneycarroll peaks into the `instance` in some of his libs though... I've marked it as private because it wasn't documented at all previously, but we may revise this...